types/react/index.d.ts

```diff
@@ -808,6 +808,8 @@ declare namespace React {
         // Form Events
         onChange?: FormEventHandler<T>;
         onChangeCapture?: FormEventHandler<T>;
+        onBeforeInput?: FormEventHandler<T>;
+        onBeforeInputCapture?: FormEventHandler<T>;
```

Comment:
oof, I can't even find out which browsers support this by searching online. Seems to be completely documented outside the uievents spec?

Reply:
At any rate, React supports it where available.